iothub/device/src/Transport/Mqtt/MqttTransportHandler.cs

```diff
@@ -4,1310 +4,1097 @@
+            {MessageSystemPropertyNames.ComponentName,IotHubWirePropertyNames.ComponentName }
+        };
+
+        internal MqttTransportHandler(PipelineContext context, IotHubClientMqttSettings settings)
+            : base(context, settings)
+        {
+            _mqttTransportSettings = settings;
+            _deviceId = context.IotHubConnectionCredentials.DeviceId;
+            _moduleId = context.IotHubConnectionCredentials.ModuleId;
+
             _methodListener = context.MethodCallback;
-            _onDesiredStatePatchListener = context.DesiredPropertyUpdateCallback;
-            _moduleMessageReceivedListener = context.ModuleEventCallback;
             _deviceMessageReceivedListener = context.DeviceEventCallback;
-        }
+            _moduleMessageReceivedListener = context.ModuleEventCallback;
+            _onDesiredStatePatchListener = context.DesiredPropertyUpdateCallback;
 
-        internal MqttTransportHandler(
-            PipelineContext context,
-            IotHubClientMqttSettings settings,
-            Func<IPAddress[], int, Task<IChannel>> channelFactory)
-            : base(context, settings)
-        {
-            _mqttIotHubAdapterFactory = new MqttIotHubAdapterFactory(settings);
-            _messageQueue = new ConcurrentQueue<Message>();
-            _completionQueue = new Queue<string>();
+            _deviceToCloudMessagesTopic = string.Format(CultureInfo.InvariantCulture, DeviceToCloudMessagesTopicFormat, _deviceId);
+            _moduleToCloudMessagesTopic = string.Format(CultureInfo.InvariantCulture, ModuleToCloudMessagesTopicFormat, _deviceId, _moduleId);
+            _deviceBoundMessagesTopic = string.Format(CultureInfo.InvariantCulture, DeviceBoundMessagesTopicFormat, _deviceId);
+            _moduleEventMessageTopic = string.Format(CultureInfo.InvariantCulture, ModuleEventMessageTopicFormat, _deviceId, _moduleId);
+            _edgeModuleInputEventsTopic = string.Format(CultureInfo.InvariantCulture, EdgeModuleInputEventsTopicFormat, _deviceId, _moduleId);
+
+            _modelId = context.ModelId;
+            _productInfo = context.ProductInfo;
+
+            var mqttFactory = new MqttFactory(new MqttLogger());
 
-            IotHubConnectionCredentials connectionCredentials = context.IotHubConnectionCredentials;
-            _additionalClientInformation = new AdditionalClientInformation
+            _mqttClient = mqttFactory.CreateMqttClient();
+            _mqttClientOptionsBuilder = new MqttClientOptionsBuilder();
+
+            _hostName = context.IotHubConnectionCredentials.IotHubHostName;
+            _connectionCredentials = context.IotHubConnectionCredentials;
+
+            if (_mqttTransportSettings.Protocol == IotHubClientTransportProtocol.WebSocket)
             {
-                ProductInfo = context.ProductInfo,
-                ModelId = context.ModelId,
-            };
+                var uri = $"wss://{_hostName}/$iothub/websocket";
```

Comment:
Would this be a place to use `string` instead of `var`? I guess the type is pretty explicit already, so it might be a subjective point.

Reply:
It is pretty obviously a string on the right side, so I'd say no.